de.metas.business/src/main/java/org/adempiere/mm/attributes/api/ImmutableAttributeSet.java

```diff
@@ -112,11 +112,12 @@ public static final ImmutableAttributeSet ofValuesIndexByAttributeId(
 			return EMPTY;
 		}
 
-		final ImmutableMap.Builder<Integer, I_M_Attribute> attributes = ImmutableMap.builder();
-		final ImmutableMap.Builder<String, I_M_Attribute> attributesByKey = ImmutableMap.builder();
-		final ImmutableMap.Builder<String, Object> valuesByAttributeKey = ImmutableMap.builder();
+		final HashMap<Integer, I_M_Attribute> attributes = new HashMap<Integer, I_M_Attribute>();
```

Comment:
hi @metas-rc 
* afaui, only the attribute values might be null. please check and use ImmutableMap where you can.
* also, afaik you can write just `new HashMap<>();` Please check your save-sactions, they should look like this: 
![image](https://user-images.githubusercontent.com/1264789/42257553-4ce76a32-7f57-11e8-921c-6a0e78bf44c4.png)